meta-facebook/yv4-sd/src/platform/plat_pldm_fw_update.c

```diff
@@ -150,32 +150,65 @@ uint8_t plat_pldm_query_device_identifiers(const uint8_t *buf, uint16_t len, uin
 		(struct pldm_query_device_identifiers_resp *)resp;
 
 	resp_p->completion_code = PLDM_SUCCESS;
-	resp_p->descriptor_count = 0x01;
+	resp_p->descriptor_count = 0x02;
 
 	uint8_t iana[PLDM_FWUP_IANA_ENTERPRISE_ID_LENGTH] = { 0x00, 0x00, 0xA0, 0x15 };
 
+	// Set the device id for sd bic
+	uint8_t deviceId[PLDM_FWUP_IANA_ENTERPRISE_ID_LENGTH] = { 0x00, 0x00 };
+
+	uint8_t total_size_of_iana_descriptor =
+		sizeof(struct pldm_descriptor_tlv) + sizeof(iana) - 1;
+
+	uint8_t total_size_of_device_id_descriptor =
+		sizeof(struct pldm_descriptor_tlv) + sizeof(deviceId) - 1;
```

Comment:
I don't understand why we're using a `uint8_t` to store these.  That might get an unexpected roll over if these structures ever grow such that `pldm_descriptor_tlv` + `deviceId` is more than 255.

Often, over specifying the type like this can also lead to *worse* code generation.  ARM arithmetic instructions are aimed at 32-bit types.  If you do arithmetic on 8-bit values, likely the compiler has to emit additional instructions to restrict the values.

I'd tend to recommend `size_t` for any sized or countable properties (variables).